relay-server/src/utils/kafka.rs

```diff
@@ -0,0 +1,20 @@
+use rdkafka::producer::{DeliveryResult, ProducerContext};
+use rdkafka::ClientContext;
+
+pub struct CaptureErrorContext;
+
+impl ClientContext for CaptureErrorContext {}
+
+impl ProducerContext for CaptureErrorContext {
+    type DeliveryOpaque = ();
+    fn delivery(&self, result: &DeliveryResult, _delivery_opaque: Self::DeliveryOpaque) {
+        if let Err((e, _message)) = result {
+            log::error!("producer error: {}", e);
```

Comment:
Not sure if it works, but could you please try to import and use `LogError` here? This ensures that we receive stack traces.

```suggestion
            log::error!("producer error: {}", LogError(e));
```

Reply:
👍